walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -62,33 +65,40 @@
       TimeRange endRange = unnestedConflicts.get(endIndex - 1);
       TimeRange overlap = TimeRange.fromStartEnd(startRange.start(), endRange.end(), false);
 
-      conflictingTimes.add(overlap);
+      builder.add(overlap);
 
       startIndex = endIndex;
     }
 
-    return ImmutableList.copyOf(conflictingTimes);
+    return builder.build();
   }
 
-  private ImmutableList<TimeRange> removeNestedTimeRanges(List<TimeRange> conflicts) {
-    List<TimeRange> unnestedConflicts = new ArrayList<>();
+  /* This method requires that conflicts be sorted by start time, then by reverse end time. */
+  private ImmutableList<TimeRange> removeNestedTimeRanges(ImmutableList<TimeRange> conflicts) {
+    ImmutableList.Builder<TimeRange> builder = new ImmutableList.Builder<TimeRange>();
 
     if (conflicts.isEmpty()) {
-      return ImmutableList.copyOf(unnestedConflicts);
+      return builder.build();
     }
 
-    unnestedConflicts.add(conflicts.get(0));
+    TimeRange lastAdded = conflicts.get(0);
+    builder.add(lastAdded);
 
     for (TimeRange conflict : conflicts) {
-      if (!unnestedConflicts.get(unnestedConflicts.size() - 1).contains(conflict)) {
-        unnestedConflicts.add(conflict);
+      if (!lastAdded.contains(conflict)) {
+        builder.add(conflict);
+        lastAdded = conflict;
       }
     }
 
-    return ImmutableList.copyOf(unnestedConflicts);
+    return builder.build();
   }
 
-  private int findEndOfOverlap(List<TimeRange> conflicts, int startIndex) {
+  /*
+   * This method finds the chain of overlapping TimeRanges in conflicts that starts at startIndex,
+   * then returns the index of the last TimeRange in that chain.
```

Comment:
I think this method is returning the index after the last TimeRange in that chain right?

Reply:
Yeah, I mixed up this implementation and my original implementation.